src/main/kotlin/no/nav/familie/ba/sak/kjerne/vedtak/vedtaksperiode/VedtaksperiodeMedBegrunnelserMigrering.kt

```diff
@@ -0,0 +1,84 @@
+package no.nav.familie.ba.sak.kjerne.vedtak.vedtaksperiode
+
+import no.nav.familie.ba.sak.common.EnvService
+import no.nav.familie.ba.sak.kjerne.behandling.domene.BehandlingRepository
+import no.nav.familie.ba.sak.kjerne.behandling.domene.BehandlingResultat
+import no.nav.familie.ba.sak.kjerne.behandling.domene.BehandlingStatus
+import no.nav.familie.ba.sak.kjerne.vedtak.VedtakService
+import org.slf4j.LoggerFactory
+import org.springframework.scheduling.annotation.Scheduled
+import org.springframework.stereotype.Component
+import org.springframework.transaction.annotation.Transactional
+import java.net.InetAddress
+import java.net.URI
+import java.net.http.HttpClient
+import java.net.http.HttpRequest
+import java.net.http.HttpResponse
+
+// TODO: Fjern etter migrering
+@Component
+class VedtaksperioderMedBegrunnelserMigrering(
+        private val behandlingRepository: BehandlingRepository,
+        private val envService: EnvService,
+        private val vedtakService: VedtakService,
+) {
+
+    @Transactional
+    @Scheduled(initialDelay = 120000, fixedDelay = Long.MAX_VALUE)
+    fun migrer() {
+        val erLeader = if (envService.erDev()) true else {
+            val client = HttpClient.newHttpClient()
+            val request = HttpRequest.newBuilder()
+                    .uri(URI.create("http://localhost:4040"))
+                    .GET()
+                    .build()
+
+            val response: HttpResponse<String> = client.send(request, HttpResponse.BodyHandlers.ofString())
+            val hostname: String = InetAddress.getLocalHost().hostName
+            logger.info("Respons=${response.body()}, hostname=$hostname")
+
+            response.body().contains(hostname)
+        }
+
+        logger.info("Er leader: $erLeader")
+
+        if (erLeader) {
+            logger.info("Migrerer behandlinger for ny begrunnelsesmodell - finn alle behandlinger under utredning.")
+            val behandlinger = behandlingRepository.finnBehandlingerForMigreringAvVedtaksbegrunnelser()
+
+            var vellykkedeMigreringer = 0
+            var mislykkedeMigreringer = 0
+            behandlinger.forEach { behandling ->
+                Result.runCatching {
+                    when {
+                        behandling.resultat == BehandlingResultat.FORTSATT_INNVILGET -> {
+                            logger.info("Hopper over behandling ${behandling.id} med resultat fortsatt innvilget")
+                        }
+                        behandling.status == BehandlingStatus.UTREDES -> {
+                            logger.info("Sett behandling ${behandling.id} til steg vilkårsvurdering.")
+                            vedtakService.settStegSlettVedtakBegrunnelserOgTilbakekreving(behandlingId = behandling.id)
+                        }
+                    }
```

Comment:
```suggestion
                            logger.info("Sett behandling ${behandling.id} til steg vilkårsvurdering.")
                            vedtakService.settStegSlettVedtakBegrunnelserOgTilbakekreving(behandlingId = behandling.id)
```

har ingenting å si, men her vil vel runcatching alltid gjøre dette?

Reply:
Enig i det, jeg tar in commiten.